src/envoy/http/jwt_auth/jwt_authenticator.cc

```diff
@@ -195,8 +195,10 @@ void JwtAuthenticator::VerifyKey(const PubkeyCacheItem& issuer_item) {
     return;
   }
 
-  // TODO: check forward_payload_header.
-  headers_->addReferenceKey(kJwtPayloadKey, jwt_->PayloadStrBase64Url());
+  forward_payload_header_.reset(
+      new LowerCaseString(issuer_item.jwt_config().forward_payload_header()));
+  headers_->addReferenceKey(*forward_payload_header_.get(),
```

Comment:
check if it is empty, if so, not to forward.

Reply:
YOu can also remove the definition of kJwtPayloadKey

Reply:
Added the check of whether the content is empty. 
For kJwtPayloadKey, Istio authn and other code that reads the output from jwt_auth have dependency on JwtAuthenticator::JwtPayloadKey(). Removing it in this PR will break their tests.  I have added a TODO to remove kJwtPayloadKey after adding the support of forward_payload_header to Istio authn and other code that reads the output from jwt_auth.